### PR TITLE
Feature/initial flow

### PR DESCRIPTION
**Pull request by dvaleda**
Setup project structure and added agents.

### CODE REVIEW

File: services/rule_engine/src/main/resources/rules/Ava/Agents.drl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package rules.Ava;
```
Comment (mjuraj) [inline services/rule_engine/src/main/resources/rules/Ava/Agents.drl]: Let's delete this file

File: services/rule_engine/src/main/resources/rules/Ava/Conversation.drl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import com.mindsmiths.armory.events.UserConnectedEvent
+import com.mindsmiths.armory.events.SubmitEvent
+
+rule "Sending pairs"
+    salience 100
+    when
+        initialize: Initialize() from entry-point "signals"
+        agent: Ava()
+    then
+        ArrayList<Integer> freeDays = new ArrayList<>();
```
Comment (mjuraj) [inline services/rule_engine/src/main/resources/rules/Ava/Conversation.drl]: Let's delete this for now

File: services/rule_engine/src/main/resources/rules/CultureMaster/Heartbeat.drl [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    when
+        initialize: Initialize() from entry-point "agent-created"
+        agent: Ava()
+    then
+        agent.sendMessage("Okay, turning my engines on. Let's start!");
+        agent.send("AVA", initialize, "new-agent");
+        delete(message);
+end
+
+rule "Get available days"
```
Comment (mjuraj) [inline services/rule_engine/src/main/resources/rules/CultureMaster/Heartbeat.drl]: Let's delete this for now

File: services/rule_engine/src/main/java/signals/DayChoiceSignal.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.mindsmiths.sdk.core.api.Signal;
+
+import lombok.Data;
+
+@Data
+public class DayChoiceSignal extends Signal {
+    ArrayList<Integer> freeDays;
+
+    public DayChoiceSignal(ArrayList<Integer> freeDays) {
```
Comment (assert-0) [inline services/rule_engine/src/main/java/signals/DayChoiceSignal.java]: Replace this constructor with Lombok's `@AllArgsConstructor`.

**Review by assert-0 — APPROVED** [review]

**Review by mjuraj — APPROVED** [review]

**Pull request merged**